app/src/androidTest/java/ch/epfl/sdp/peakar/camera/CameraActivityTest.java

```diff
@@ -55,27 +97,50 @@ public void cleanUp(){
         Intents.release();
     }
 
+    /* Test that the compass is changed when clicking on the compass button */
+    @Test
+    public void TestChangeCompassButton() throws InterruptedException {
+        String displayCompassKey = context.getResources().getString(R.string.displayCompass_key);
+        sharedPreferences.edit().putBoolean(displayCompassKey, false).apply();
+
+        ViewInteraction button = Espresso.onView(withId(R.id.compassMiniature));
+        button.perform(ViewActions.click());
+
+        Thread.sleep(SHORT_SLEEP_TIME);
+
+        boolean displayCompass = sharedPreferences.getBoolean(displayCompassKey, false);
+        assertTrue(displayCompass);
+    }
 
-    /* Test that pressing the map icon button changes view to MapActivity */
+    /* Test that pressing the profile button when signed-out launches the ProfileLaunchActivity */
     @Test
-    public void TestMapIconButton(){
-        ViewInteraction button = Espresso.onView(ViewMatchers.withId(R.id.mapButton));
+    public void TestProfileButtonNotSignedIn(){
+        removeAuthUser();
+        ViewInteraction button = Espresso.onView(withId(R.id.profileButton));
         button.perform(ViewActions.click());
-        // Catch intent
-        intended(IntentMatchers.hasComponent(MapActivity.class.getName()));
+        intended(IntentMatchers.hasComponent(ProfileLauncherActivity.class.getName()));
     }
 
-    /* Test that pressing the back button finish the activity */
+    /* Test that pressing the profile button when signed in launches the ProfileLaunchActivity */
     @Test
-    public void TestBackButton(){
-        ViewInteraction button = Espresso.onView(withId(R.id.cameraBackButton));
+    public void TestProfileButtonSignedIn(){
+
+        createTestUser();
+
+        ViewInteraction button = Espresso.onView(withId(R.id.profileButton));
         button.perform(ViewActions.click());
-        try {
-            Thread.sleep(1000);
-            assertSame(testRule.getScenario().getResult().getResultCode(), Activity.RESULT_CANCELED);
-        } catch (InterruptedException e) {
-            e.printStackTrace();
-            fail("TestBackButton failed");
+
+        intended(IntentMatchers.hasComponent(ProfileActivity.class.getName()));
+    }
+
+    public void createTestUser() {
+        if(FirebaseAuth.getInstance().getCurrentUser() == null) {
+            AuthService.getInstance().signOut(InstrumentationRegistry.getInstrumentation().getTargetContext());
+            registerAuthUser();
+        }
+        else {
+            FirebaseAuthService.getInstance().forceRetrieveData();
         }
```

Comment:
In my PR I've put these lines inside the `registerAuthUser()` method as they were repeated everywhere. As we don't know which PR will be merged first, should we add here a TODO comment saying that we can replace all these lines with just a `registerAuthUser()`?

Reply:
Yes, I added TODO comments.